src/SimpleAmqpClient/Channel.h

```diff
@@ -89,9 +90,11 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable
                         const std::string &username = "guest",
                         const std::string &password = "guest",
                         const std::string &vhost = "/",
-                        int frame_max = 131072)
+                        int frame_max = 131072,
+                        const amqp_sasl_method_enum sasl_method=AMQP_SASL_METHOD_PLAIN)
```

Comment:
amqp_login_with_properties only needs username and password when using the PLAIN SASL method. Those parameters aren't required for the EXTERNAL method.